godot-core/src/meta/args/as_arg.rs

```diff
@@ -53,16 +53,34 @@ use std::ffi::CStr;
     note = "GString/StringName/NodePath aren't implicitly convertible for performance reasons; use their `arg()` method.",
     note = "see also `AsArg` docs: https://godot-rust.github.io/docs/gdext/master/godot/meta/trait.AsArg.html"
 )]
-pub trait AsArg<T: ParamType>
+pub trait AsArg<T: ToGodot>
 where
     Self: Sized,
 {
     #[doc(hidden)]
-    fn into_arg<'r>(self) -> <T as ParamType>::Arg<'r>
+    fn into_arg<'r>(self) -> CowArg<'r, T>
```

Comment:
It is noteworthy that while this change is possible, `CowArg` introduces a small runtime penalty.

In particular, `Copy` types like `i64`, `bool`, `Vector2i` etc. now require a bigger size (for the discriminant) and case differentiation (borrowed vs. owned) at runtime, although they're always owned. 

Luckily the `AsArg` trait isn't used in generated engine APIs for `Copy` types, but it is present in APIs like `Array::push()` or `<GeneratedSignal>::emit()`. In these it's probably OK due to the much bigger overhead of variant marshalling or signal book-keeping :slightly_smiling_face:

Reply:
I added a comment.